jar-async/src/slipstream/async/collector.clj

```diff
@@ -159,53 +162,77 @@
 (defonce ^:dynamic *all-collect-processor* (collect-readers all-collector-chan))
 
 (defn add-increasing-space
- [ucs space]
- (let [spaces (iterate #(+ space %) 0)]
+ [ucs step]
+ (let [spaces (iterate #(+ step %) 0)]
    (map conj ucs spaces)))
 
-(defn warn-channel-full   
+(defn warn-channel-full
   [context u c]
-  (log/log-error context 
+  (log/log-error context
     ": the processing channel capacity(" collector-chan-size") is currently full, will *not* insert request for "
       (user-connector u c)))
 
-(defn inform-avoiding   
+(defn inform-avoiding
   [context u c]
   (log/log-info context ": avoiding inserting on " (user-connector u c) " being already processed."))
 
-(defn inform-inserted 
+(defn inform-inserted
   [context u c t]
-  (log/log-debug context ": inserted collect request for " (user-connector u c) " after timeout " t))    
+  (log/log-debug context ": inserted collect request for " (user-connector u c) " after timeout " t))
 
-(defn inform-nothing-to-do 
+(defn inform-nothing-to-do
   [context]
   (log/log-debug context ": no users to collect."))
 
-(defn show-current-channel-usage   
+(defn show-current-channel-usage
   [context]
   (let [[current total ratio] (current-working-usage)]
-    (log/log-info context ": " current " / " total " => " ratio " %")))
+    (log/log-info "indirect channel usage before inserting collect requests for " context ": " current " / " total " => " ratio " %")))
+
+(defn log-initial-workspace
+  [nu nc]
+  (log/log-debug "Requested work on #" nu " users and #" nc " connectors. Initial workspace: " (* nu nc)))
+
+(defn log-final-workspace
+  [n]
+  (log/log-debug "After filtering out users w/o connector creds. Final workspace: " n))
+
+(defn log-spread
+  [s t]
+  (log/log-debug "Will spread " s " requests in " (msecs-in-seconds t) " s."))
+
+(defn insert-collection-request-cond-spaced
+  [chan u c t context]
+  (go
+    (<! (timeout t))
+    (if-not (busy? u c)
+      (do
+        (busy! u c)
+        (>! chan [u c])
+        (inform-inserted context u c t))
+      (inform-avoiding context u c))))
+
+(defn compose-uc-pairs
+  [users connectors]
+  (remove nil?
+          (for [u users c connectors]
+            (when (.isCredentialsSet c u) [u c]))))
 
```

Comment:
minor:
`remove nil?` can be avoided by using `:when` inside `for`
 (for [u users c connectors] :when (.isCredentialsSet c u) [u c]))